Library/Homebrew/software_spec.rb

```diff
@@ -336,20 +336,18 @@ def checksum_for(tag)
   end
 
   def checksums
-    checksums = {}
-    os_versions = collector.keys
-    os_versions.map! do |macos|
+    tags = collector.keys.sort_by do |tag|
       begin
-        MacOS::Version.from_symbol macos
+        MacOS::Version.from_symbol tag
       rescue
-        nil
+        "0.#{tag}"
```

Comment:
It looks like this means it's changing the actual tag that's returned by this function, though, which doesn't seem to be desirable. I think the higher-level refactoring you mentioned earlier would be preferable; this reads like a bit of a "hack to make it work" unfortunately.